apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -1111,5 +1119,103 @@ public void AckModifyAckDeadlineFault_Thrown([CombinatorialValues(true, false)]
                 Assert.Equal(1, fake.ClientShutdowns.Count);
             });
         }
+
+        [Theory, CombinatorialData]
+        public void ExactlyOnce_TemporaryFault([CombinatorialValues(true, false, null)] bool? ackNackOrExtends)
+        {
+            var msgs = new[]
+            {
+                ServerAction.Data(TimeSpan.Zero, new[] { "1" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "2" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "3" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "4" }),
+                ServerAction.Inf()
+            };
+
+            // Any temporary failure in ack/nack/extend request should not be thrown to the client. It should be retried.
+            var exception = new RpcException(new Status(StatusCode.DeadlineExceeded, ""), "");
+            // If ackNackOrExtends is null, then it is extend request. ModifyAcknowledgeDeadline RPC will throw the supplied exception.
+            // If ackNackOrExtends is true, then it is acknowledge request. Acknowledge RPC will throw the supplied exception.
+            // If ackNackOrExtends is false, then it is nack request. ModifyAcknowledgeDeadline RPC will throw the supplied exception.
+            var ackModifyAckDeadlineAction =
+                ackNackOrExtends == null ? AckModifyAckDeadlineAction.Data(exception, 10, false, true)
```

Comment:
Maybe use named arguments for 10, false, true so we have some idea what they mean?

Reply:
Done.